File: frontend/src/objects/problem/response/ProblemDataTreeResponse.ts
import type { ProblemDataTreeNode } from '@/objects/problem/response/ProblemDataTreeNode'

/** 题目数据树响应；items 为后端整理后的扁平节点集合。 */
export type ProblemDataTreeResponse = {
  items: ProblemDataTreeNode[]
}

File: backend/src/main/scala/domains/problem/objects/response/ProblemDataTreeResponse.scala
package domains.problem.objects.response

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 题目数据树响应；items 已经包含目录和文件节点，由后端排序生成。 */
final case class ProblemDataTreeResponse(items: List[ProblemDataTreeNode])

/** ProblemDataTreeResponse 的 JSON 编解码器。 */
object ProblemDataTreeResponse:
  given Encoder[ProblemDataTreeResponse] = deriveEncoder[ProblemDataTreeResponse]
  given Decoder[ProblemDataTreeResponse] = deriveDecoder[ProblemDataTreeResponse]
